Snakefile

```diff
@@ -1,6 +1,356 @@
+    conda: "envs/isl.yaml"
+    log: f"logs/irap_stage0_{FIRST_SAMPLE}.log"
+    params:
+        private_script=config["private_script"],
+        conf=config["irap_config"],
+        root_dir=config["atlas_gtex_root"],
+        strand="both",
+        irapMem=4096000000,
+        irapDataOption="",
+        filename= f"{FIRST_SAMPLE}",
+        read_type = detect_read_type
+    resources: mem_mb=10000
+    threads: 16
+    shell:
+        """
+        set -e # snakemake on the cluster doesn't stop on error when --keep-going is set
+        exec &> "{log}"
+		
+        source {params.private_script}/gtex_bulk_env.sh
+        source {params.private_script}/gtex_bulk_init.sh
+        #source {params.root_dir}/isl/lib/functions.sh
+        source {params.root_dir}/isl/lib/generic_routines.sh
+        source {params.root_dir}/isl/lib/process_routines.sh
+        source {params.root_dir}/isl/lib/irap.sh
+	
+        cp {params.private_script}/gtex_bulk_env.sh $IRAP_SINGLE_LIB
+	
+        cat {input.check}
+
+        library={params.filename}
+        echo "library: $library"
+        workingDir=$ISL_WORKING_DIR
+
+        source {params.root_dir}/scripts/aux.sh
+
+        which get_local_relative_library_path
+        which get_library_path
+
+
+        localFastqPath=$(get_local_relative_library_path $library )
+	
+        echo "workingDir: $workingDir"
+        echo "localFastqPath: $localFastqPath"
+        
+        mkdir -p $(dirname $workingDir/$localFastqPath)
+
+        pushd $workingDir > /dev/null
+	
+        if [[ {params.read_type} == "se" ]]; then
+            # fastq is SE
+            cp {input.fastq} $workingDir/${{localFastqPath}}.fastq
+
+            echo "Calling irap_single_lib...SE mode"
+            cmd="irap_single_lib -0 -A -f -o irap_single_lib -1 $workingDir/${{localFastqPath}}.fastq -c {params.conf} -s {params.strand} -m {params.irapMem} -t {threads} -C {params.irapDataOption}"
+            echo "stage0 will run now:"
+            eval $cmd
+            echo "stage0 finished"
+        else
+            # fastq is PE
+            #split_fastq {input.fastq} $workingDir ${{localFastqPath}}
+            reformat.sh ow=t int=t vpair=t vint=t in={input.fastq} out1=$workingDir/${{localFastqPath}}_1.fastq out2=$workingDir/${{localFastqPath}}_2.fastq
+
+            echo "Calling irap_single_lib...PE mode"
+            cmd="irap_single_lib -0 -A -f -o irap_single_lib -1 ${{localFastqPath}}_1.fastq -2 ${{localFastqPath}}_2.fastq -c {params.conf} -s {params.strand} -m {params.irapMem} -t {threads} -C {params.irapDataOption}"
+            echo "stage0 will run now:"
+            eval $cmd
+            echo "stage0 finished"
+        fi
+
+        popd
+
+        touch {output}
+        """
+
+
+rule run_irap:
+    input:
+        fastq=rules.bam_to_fastq.output.fastq,
+        check = rules.validating_fastq.output.val_fastq,
+        stage0_completed=rules.run_irap_stage0.output
+    output: "out/{sample}.txt"
```

Comment:
This was only for testing purpose this can be replaced with one of the iRAP output..